crates/api_models/src/events/payment.rs

```diff
@@ -196,13 +196,26 @@ impl ApiEventMetric for PaymentsResponse {
 }
 
 impl ApiEventMetric for PaymentMethodResponse {
+    #[cfg(all(
+        any(feature = "v1", feature = "v2"),
+        not(feature = "payment_methods_v2")
+    ))]
     fn get_api_event_type(&self) -> Option<ApiEventsType> {
         Some(ApiEventsType::PaymentMethod {
             payment_method_id: self.payment_method_id.clone(),
             payment_method: self.payment_method,
             payment_method_type: self.payment_method_type,
         })
     }
+
+    #[cfg(all(feature = "v2", feature = "payment_methods_v2"))]
+    fn get_api_event_type(&self) -> Option<ApiEventsType> {
+        Some(ApiEventsType::PaymentMethod {
+            payment_method_id: self.payment_method_id.clone(),
+            payment_method: self.payment_method_type,
```

Comment:
are we not changing the field name her?

Reply:
Since renaming the field affects the analytics pipeline, we'd have to include ClickHouse migrations as well.
For now, I'm not including v2 changes in the analytics pipeline.

CC: @ShivanshMathurJuspay @tsdk02

Reply:
Are we renaming payment_method to payment_method_type ?